code/modules/projectiles/projectile.dm

```diff
@@ -302,6 +302,21 @@
 				Range()
 			sleep(1)
 
+obj/item/projectile/proc/reflect_back(atom/source, list/position_modifiers = list(0, 0, 0, 0, 0, -1, 1, -2, 2))
+	if(starting)
+		var/new_x = starting.x + pick(position_modifiers)
+		var/new_y = starting.y + pick(position_modifiers)
+		var/turf/curloc = get_turf(source)
+
+		// redirect the projectile
+		firer = source
+		original = locate(new_x, new_y, z)
+		starting = curloc
+		current = curloc
+		yo = new_y - curloc.y
+		xo = new_x - curloc.x
+		Angle = null // Will be calculated in fire()

```

Comment:
Was the missing part of the construct code which meant that it didn't change the target of the projectile.